data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/BeaconRestApi.java

```diff
@@ -75,6 +80,70 @@ public void start() {
                     ctx.status(SC_NOT_FOUND).result(JsonProvider.objectToJSON("Not found"));
                   }
                 }));
+
+    app.start();
+  }
+
+  private OpenApiOptions getOpenApiOptions() {
+    // TODO: Need real values for version etc
+    Info applicationInfo =
+        new Info()
+            .version("0.1.0")
+            .title("Minimal Beacon Node API for Validator")
+            .description(
+                "A minimal API specification for the beacon node, which enables a validator "
+                    + "to connect and perform its obligations on the Ethereum 2.0 phase 0 beacon chain.")
+            .license(
+                new License()
+                    .name("Apache 2.0")
+                    .url("https://www.apache.org/licenses/LICENSE-2.0.html"));
+    OpenApiOptions options =
+        new OpenApiOptions(applicationInfo)
+            .activateAnnotationScanningFor("tech.pegasys.artemis.beaconrestapi")
+            .path("/swagger-docs")
+            .swagger(new SwaggerOptions("/swagger-ui"));
+    // TODO: allow swagger-ui to be turned off - ideally still leave swagger-docs, just dont add the
+    // swagger-ui endpoint
+    return options;
+  }
+
+  private void addNodeHandlers() {
+    app.get(GenesisTimeHandler.ROUTE, new GenesisTimeHandler(chainStorageClient));
+    /*
+     * TODO:
+     *    /node/version
+     *    /node/syncing
+     *  Optional:
+     *    /node/fork
+     */
+  }
+
+  private void addBeaconHandlers() {
+    // TODO: not in Minimal or optional specified set - some are similar to lighthouse
```

Comment:
Very recently I've added beaconChainHead and modified beaconBlock to expose Alethio a similar API to Prysm's. FYI. It would be nice to connect you guys actually (with Alethio) since they are going to be using this API a lot.

Reply:
absolutely! Arash is coordinating as PO, we should get together and understand where things are at.